src/cfile/core.py

```diff
@@ -197,17 +236,13 @@ class Struct(DataType):
     """
     A struct definition
     """
-    def __init__(self, name: str | None, members: StructMember | list[StructMember] | None = None) -> None:
+    def __init__(self, name: str = "", members: list[StructMember] = [], attributes: list[str] = []) -> None:
         super().__init__(name)
-        self.members: list[StructMember] = []
-        if members is not None:
-            if isinstance(members, StructMember):
-                self.append(members)
-            elif isinstance(members, list):
-                for member in members:
-                    self.append(member)
-            else:
-                raise TypeError('Invalid argument type for "elements"')
+        if isinstance(members, list):
+            self.members: list[StructMember] = members.copy()
+        else:
+            raise TypeError('Invalid argument type for "elements"')
+        self.attributes = attributes.copy()
```

Comment:
Don't use copy. If you want to create a new list which is a copy of another, write it as `list(attributes)`.
But as stated above, we should introduce attribute-mechanism later anyway.